benchmark_runner/common/oc/oc.py

```diff
@@ -457,32 +457,51 @@ def get_worker_nodes(self):
         """
         return self.run(fr""" {self.__cli} get nodes -l node-role.kubernetes.io/worker= -o jsonpath="{{range .items[*]}}{{.metadata.name}}{{'\n'}}{{end}}" """)
 
-    @staticmethod
     @typechecked
-    def check_node_status(nodes_list: list):
+    def wait_for_nodes_ready(self, wait_time: int = None, timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
         """
-        This method check node status
-        @param nodes_list:
-        @return: True when all nodes in ready status
+        This method waits until all nodes are in 'Ready' status.
+        @param wait_time: wait time between each loop
+        @param timeout: Maximum wait time in seconds (default set in environment variables)
+        @return: True if all nodes are in 'Ready' status within the timeout period
+        @raises: NodeNotReady if one or more nodes are not ready within the timeout
+        """
+        wait_time = wait_time or OC.SHORT_TIMEOUT
+        all_nodes_ready = None
+        current_wait_time = 0
+        while timeout <= 0 or current_wait_time < timeout:
+            all_nodes_ready = self.check_all_nodes_status()
+            if all_nodes_ready is True:
+                return True
+            logger.info(f"Waiting for node '{all_nodes_ready[0]}' to reach 'Ready' status, current status '{all_nodes_ready[1]}'.")
+            time.sleep(wait_time)
+            current_wait_time += wait_time
+        logger.info(f"oc get nodes:\n{self.run('oc get nodes')}")
+        raise NodeNotReady(node_name=all_nodes_ready[0], node_status=all_nodes_ready[1])
+
+    @typechecked
+    def check_all_nodes_status(self):
+        """
+        This method checks the status of all nodes in the list.
+        @return: True if all nodes are in 'Ready' status; otherwise, return (node_name, node_status)
         """
         # Check if any node is not in 'Ready' status
-        for node in nodes_list:
+        for node in self.get_node_status():
             node_name, node_status = node.split()
             if node_status != 'Ready':
-                raise NodeNotReady(node_name, node_status)
+                return node_name, node_status
```

Comment:
Return all nodes that are not ready, not just the first one.

Reply:
Good catch, add dictionary to hold not ready node: `not_ready_nodes = {}`